### ISSUE TITLE
wrong arguments to torch.full_like(self.temperature, float(K)) in ExpRelaxedCategorical.log_prob 

### ISSUE THREAD

**Issue by tkusmierczyk** [pytorch/pytorch#31825]
## Wrong arguments to torch.full_like(self.temperature, float(K)) in ExpRelaxedCategorical.log_prob 

Using log_prob of ExpRelaxedCategorical distribution regardless of arguments raises:
full_like() received an invalid combination of arguments - got (int, float), but expected one of:
 * (Tensor input, Number fill_value, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 * (Tensor input, Number fill_value, bool requires_grad)
because full_like(.) receives wrong arguments:
 - self.temperature is a float
 - K is an integer

## To Reproduce

import torch
dist = torch.distributions.relaxed_categorical.ExpRelaxedCategorical(0.1, logits=torch.tensor([0.1,0.2,0.3], dtype=torch.float32) )
dist.log_prob(5)

## Environment

PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.16.2
[pip3] numpydoc==0.7.0
[conda] _pytorch_select           0.2                       gpu_0  
[conda] _tflow_select             2.3.0                       mkl  
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      243  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.0.15           py37ha843d7b_0  
[conda] mkl_random                1.1.0            py37hd6b4f25_0  
[conda] pytorch                   1.3.1           cuda100py37h53c1284_0  
[conda] tensorflow                2.0.0           mkl_py37h66b46cc_0  
[conda] tensorflow-base           2.0.0           mkl_py37h9204916_0

**Comment by tkusmierczyk** [comment]
temperature for a reason is required to be a tensor...
